skills/declarative/calendarSkill/runtime/customaction/Actions/MSGraph/GetContacts.cs

```diff
@@ -134,5 +136,11 @@ public override async Task<DialogTurnResult> BeginDialogAsync(DialogContext dc,
             // return the actionResult as the result of this operation
             return await dc.EndDialogAsync(result: results, cancellationToken: cancellationToken);
         }
+
+        private bool IsEmail(string emailString)
+        {
+            bool isEmail = !string.IsNullOrEmpty(emailString) && Regex.IsMatch(emailString, @"\A(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)\Z", RegexOptions.IgnoreCase);
```

Comment:
For perf gain you might want to also compile the regex as well.

Reply:
Added a RegexOptions.Compiled. Does that work?